File: lines.cs
// lines.cs
using System;
using System.Collections.Generic;

class Puzzle
{
    public int Size;
    public List<(int,int,int,int)> Pairs; // (r1,c1,r2,c2)

    public Puzzle(int size, int seed)
    {
        Size = size;
        var rand = new Random(seed);
        Pairs = new List<(int,int,int,int)>();
        if (size == 6)
        {
            var data = new (int,int,int,int)[]
            {
                (0,0,5,5), (0,1,4,3), (1,0,3,4),
                (2,2,4,4), (3,1,5,3), (0,5,5,0)
            };
            Pairs.AddRange(data);
        }
        else
        {
            for (int i=0; i<size/2; i++)
            {
                int r1=rand.Next(size), c1=rand.Next(size);
                int r2=rand.Next(size), c2=rand.Next(size);
                Pairs.Add((r1,c1,r2,c2));
            }
        }
    }

    public void Display(bool showSolution)
    {
        string letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        Console.Write("  ");
        for (int i=0; i<Size; i++) Console.Write((i+1)+" ");
        Console.WriteLine();
        for (int r=0; r<Size; r++)
        {
            Console.Write(letters[r]+" ");
            for (int c=0; c<Size; c++)
            {
                bool found = false;
                for (int idx=0; idx<Pairs.Count; idx++)
                {
                    var p = Pairs[idx];
                    if ((r==p.Item1 && c==p.Item2) || (r==p.Item3 && c==p.Item4))
                    {
                        Console.Write((idx+1)+" ");
                        found = true;
                        break;
                    }
                }
                if (!found) Console.Write(". ");
            }
            Console.WriteLine();
        }
    }
}

class Program
{
    static void Main(string[] args)
    {
        int size = 6;
        bool showSolution = false;
        for (int i=0; i<args.Length; i++)
        {
            if (args[i] == "--size" && i+1 < args.Length)
                size = int.Parse(args[++i]);
            else if (args[i] == "--show-solution")
                showSolution = true;
        }
        int seed = (int)DateTime.Now.Ticks;
        var p = new Puzzle(size, seed);
        p.Display(showSolution);
    }
}

File: Lines.java
// Lines.java
import java.util.*;

public class Lines {
    static class Puzzle {
        int size;
        List<int[][]> pairs; // каждая пара: [ [r1,c1], [r2,c2] ]
        Puzzle(int size, long seed) {
            this.size = size;
            Random rand = new Random(seed);
            pairs = new ArrayList<>();
            if (size == 6) {
                int[][][] data = {
                    {{0,0},{5,5}}, {{0,1},{4,3}}, {{1,0},{3,4}},
                    {{2,2},{4,4}}, {{3,1},{5,3}}, {{0,5},{5,0}}
                };
                for (int[][] d : data) pairs.add(d);
            } else {
                for (int i=0; i<size/2; ++i) {
                    int[][] p = {
                        {rand.nextInt(size), rand.nextInt(size)},
                        {rand.nextInt(size), rand.nextInt(size)}
                    };
                    pairs.add(p);
                }
            }
        }
        void display(boolean showSolution) {
            String letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
            System.out.print("  ");
            for (int i=0; i<size; ++i) System.out.print((i+1)+" ");
            System.out.println();
            for (int r=0; r<size; ++r) {
                System.out.print(letters.charAt(r)+" ");
                for (int c=0; c<size; ++c) {
                    boolean found = false;
                    for (int idx=0; idx<pairs.size(); ++idx) {
                        int[][] p = pairs.get(idx);
                        if ((r==p[0][0] && c==p[0][1]) ||
                            (r==p[1][0] && c==p[1][1])) {
                            System.out.print((idx+1)+" ");
                            found = true;
                            break;
                        }
                    }
                    if (!found) System.out.print(". ");
                }
                System.out.println();
            }
        }
    }

    public static void main(String[] args) {
        int size = 6;
        boolean showSolution = false;
        for (int i=0; i<args.length; ++i) {
            if (args[i].equals("--size") && i+1<args.length) {
                size = Integer.parseInt(args[++i]);
            } else if (args[i].equals("--show-solution")) {
                showSolution = true;
            }
        }
        long seed = System.currentTimeMillis();
        Puzzle p = new Puzzle(size, seed);
        p.display(showSolution);
    }
}
